### ISSUE TITLE
[Bug]: Create index failed with exception `DescribeSegments failed: cannot find index, id =xx` after shrinking indexnode

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#16832]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 6b2d3c3e
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.0.dev47
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. Deploy milvus cluster with 2 indexnodes
2. Create collection and insert data
3. Create index successfully and drop index
4. Scale in indexnode from 2 to 1
5. Create index again and failed
```
[2022-05-07 15:55:26,081 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=DescribeSegments failed: cannot find index, id = 433038323890257921)> (api_request.py:36)
```

**Server logs:**
[mic_logs.tar.gz](https://github.com/milvus-io/milvus/files/8644750/mic_logs.tar.gz)

### Expected Behavior

Creating index successfully and indexing time is about twice as long.

### Steps To Reproduce

```markdown
# deploy cluster with 2 indexnodes

            connections.add_connection(default={"host": host, "port": 19530})
            connections.connect(alias='default')

            data = cf.gen_default_dataframe_data(nb)

            # create
            c_name = "index_scale_one"
            collection_w = ApiCollectionWrapper()
            # collection_w.init_collection(name=c_name)
            collection_w.init_collection(name=c_name, schema=cf.gen_default_collection_schema())
            # insert
            loop = 10
            for i in range(loop):
                collection_w.insert(data)
            assert collection_w.num_entities == nb * loop

            # create index on collection one and two
            start = datetime.datetime.now()
            collection_w.create_index(ct.default_float_vec_field_name, default_index_params)
            assert collection_w.has_index()[0]
            t0 = datetime.datetime.now() - start

            log.info(f'Create index on 2 indexNode cost t0: {t0}')

            collection_w.drop_index()
            assert not collection_w.has_index()[0]

            # shrink indexNode from 2 to 1
            mic.upgrade(release_name, {'spec.components.indexNode.replicas': 1}, constants.NAMESPACE)
            mic.wait_for_healthy(release_name, constants.NAMESPACE)
            wait_pods_ready(constants.NAMESPACE, f"app.kubernetes.io/instance={release_name}")

            start = datetime.datetime.now()
            collection_w.create_index(ct.default_float_vec_field_name, default_index_params)
            assert collection_w.has_index()[0]
            t1 = datetime.datetime.now() - start
            log.info(f'Create index on 1 indexNode cost t1: {t1}')
            collection_w.drop_index()

```
```


### Milvus Log

```markdown
<partitions>: [{"name": "_default", "collection_name": "index_scale_one", "description": ""}]
<description>: 
<schema>: {
  auto_id: False
  description: 
  fields: [{
    name: int64
    description: 
    type: 5
    is_primary: True
    auto_id: ......  (api_request.py:27)
[2022-05-07 15:54:54,197 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float varchar                                       float_vector
0         0     0.0       0  [0.01950958175746401, 0.07034751725723987, 0.0...
1         1     1.0       1  [0.09165761673970102, 0.0639118894757802, 0.05...
2         2     2.0       2  [0.048868545614790454, 0.14946446......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-07 15:54:55,222 - DEBUG - ci_test]: (api_response) : (insert count: 5000, delete count: 0, upsert count: 0, timestamp: 433038319918776321)  (api_request.py:27)
......

[2022-05-07 15:55:05,180 - DEBUG - ci_test]: (api_request)  : [Collection.insert] args: [      int64   float varchar                                       float_vector
0         0     0.0       0  [0.01950958175746401, 0.07034751725723987, 0.0...
1         1     1.0       1  [0.09165761673970102, 0.0639118894757802, 0.05...
2         2     2.0       2  [0.048868545614790454, 0.14946446......, kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-07 15:55:06,383 - DEBUG - ci_test]: (api_response) : (insert count: 5000, delete count: 0, upsert count: 0, timestamp: 433038322867896321)  (api_request.py:27)
[2022-05-07 15:55:10,061 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['float_vector', {'index_type': 'IVF_SQ8', 'metric_type': 'L2', 'params': {'nlist': 128}}], kwargs: {'timeout': 40} (api_request.py:55)
[2022-05-07 15:55:14,894 - DEBUG - ci_test]: (api_response) : Status(code=0, message='')  (api_request.py:27)
[2022-05-07 15:55:14,895 - DEBUG - ci_test]: (api_request)  : [Collection.has_index] args: [], kwargs: {} (api_request.py:55)
[2022-05-07 15:55:14,909 - DEBUG - ci_test]: (api_response) : True  (api_request.py:27)
[2022-05-07 15:55:14,910 - INFO - ci_test]: Create index on 2 indexNode cost t0: 0:00:04.849029 (test_index_node_scale.py:171)
[2022-05-07 15:55:14,911 - DEBUG - ci_test]: (api_request)  : [Collection.drop_index] args: [], kwargs: {'timeout': 20} (api_request.py:55)
[2022-05-07 15:55:14,984 - DEBUG - ci_test]: (api_response) : None  (api_request.py:27)
[2022-05-07 15:55:14,985 - DEBUG - ci_test]: (api_request)  : [Collection.has_index] args: [], kwargs: {} (api_request.py:55)
[2022-05-07 15:55:14,998 - DEBUG - ci_test]: (api_response) : False  (api_request.py:27)
[2022-05-07 15:55:15,000 - DEBUG - ci_test]: upgrade milvus with configs: {'spec': {'components': {'indexNode': {'replicas': 1}}}} (milvus_operator.py:111)
[2022-05-07 15:55:15,131 - DEBUG - ci_test]: patch custom resource response: {'apiVersion': 'milvus.io/v1alpha1', 'kind': 'MilvusCluster', 'metadata': {'creationTimestamp': '2022-05-07T07:42:04Z', 'finalizers': ['milvuscluster.milvus.io/finalizer'], 'generation': 3, 'labels': {'app': 'milvus'}, 'managedFields': [{'apiVersion': 'milvus.io/v1alpha1', 'fieldsType': 'FieldsV1', 'fieldsV1': {'f:metadata': {'f:labels': {'.': {}, 'f:app': {}}}, 'f:spec': {'.': {}, 'f:components': {'.': {}, 'f:dataNode': {'.': {}, 'f:replicas': {}}, 'f:image': {}, 'f:indexNode': {'.': {}, 'f:replicas': {}}, 'f:proxy': {'.': {}, 'f:replicas': {}, 'f:serviceType': {}}}, 'f:config': {'.': {}, 'f:dataCoord': {'.': {}, 'f:enableCompaction': {}, 'f:enableGarbageCollection': {}}}, 'f:dependencies': {'.': {}, 'f:etcd': {'.': {}, 'f:external': {}, 'f:inCluster': {'.': {}, 'f:deletionPolicy': {}, 'f:pvcDeletion': {}}}, 'f:pulsar': {'.': {}, 'f:external': {}, 'f:inCluster': {'.': {}, 'f:deletionPolicy': {}, 'f:pvcDeletion': {}, 'f:values': {'.': {}, 'f:image': {'.': {}, 'f:bastion': {'.': {}, 'f:tag': {}}, 'f:bookkeeper': {'.': {}, 'f:tag': {}}, 'f:broker': {'.': {}, 'f:tag': {}}, 'f:proxy': {'.': {}, 'f:tag': {}}, 'f:zookeeper': {'.': {}, 'f:tag': {}}}}}}, 'f:storage': {'.': {}, 'f:external': {}, 'f:inCluster': {'.': {}, 'f:deletionPolicy': {}, 'f:pvcDeletion': {}}, 'f:type': {}}}}}, 'manager': 'OpenAPI-Generator', 'operation': 'Update', 'time': '2022-05-07T07:42:04Z'}, {'apiVersion': 'milvus.io/v1alpha1', 'fieldsType': 'FieldsV1', 'fieldsV1': {'f:metadata': {'f:finalizers': {}}, 'f:spec': {'f:dependencies': {'f:etcd': {'f:endpoints': {}}, 'f:pulsar': {'f:endpoint': {}}, 'f:storage': {'f:endpoint': {}}}}, 'f:status': {'.': {}, 'f:conditions': {}, 'f:endpoint': {}, 'f:ingress': {'.': {}, 'f:loadBalancer': {}}, 'f:replicas': {'.': {}, 'f:dataCoord': {}, 'f:dataNode': {}, 'f:indexCoord': {}, 'f:indexNode': {}, 'f:proxy': {}, 'f:queryCoord': {}, 'f:queryNode': {}, 'f:rootCoord': {}}, 'f:status': {}}}, 'manager': 'manager', 'operation': 'Update', 'time': '2022-05-07T07:45:48Z'}], 'name': 'shrink-index', 'namespace': 'chaos-testing', 'resourceVersion': '321843673', 'uid': '4d190f93-ea70-45e0-a5ee-2b373e4ea005'}, 'spec': {'components': {'dataCoord': {'replicas': 1}, 'dataNode': {'replicas': 2}, 'image': 'milvusdb/milvus-dev:master-20220507-6b2d3c3e', 'indexCoord': {'replicas': 1}, 'indexNode': {'replicas': 1}, 'proxy': {'replicas': 1, 'serviceType': 'LoadBalancer'}, 'queryCoord': {'replicas': 1}, 'queryNode': {'replicas': 1}, 'rootCoord': {'replicas': 1}}, 'config': {'dataCoord': {'enableCompaction': True, 'enableGarbageCollection': True}}, 'dependencies': {'etcd': {'endpoints': ['shrink-index-etcd.chaos-testing:2379'], 'external': False, 'inCluster': {'deletionPolicy': 'Delete', 'pvcDeletion': True, 'values': {}}}, 'pulsar': {'endpoint': 'shrink-index-pulsar-proxy.chaos-testing:6650', 'external': False, 'inCluster': {'deletionPolicy': 'Delete', 'pvcDeletion': True, 'values': {'image': {'bastion': {'tag': '2.8.2'}, 'bookkeeper': {'tag': '2.8.2'}, 'broker': {'tag': '2.8.2'}, 'proxy': {'tag': '2.8.2'}, 'zookeeper': {'tag': '2.8.2'}}}}}, 'storage': {'endpoint': 'shrink-index-minio.chaos-testing:9000', 'external': False, 'inCluster': {'deletionPolicy': 'Delete', 'pvcDeletion': True, 'values': {}}, 'secretRef': 'shrink-index-minio', 'type': 'MinIO'}}}, 'status': {'conditions': [{'lastTransitionTime': '2022-05-07T07:43:22Z', 'message': 'Etcd endpoints is healthy', 'reason': 'EtcdReady', 'status': 'True', 'type': 'EtcdReady'}, {'lastTransitionTime': '2022-05-07T07:43:22Z', 'reason': 'StorageReady', 'status': 'True', 'type': 'StorageReady'}, {'lastTransitionTime': '2022-05-07T07:45:18Z', 'message': 'Pulsar is ready', 'reason': 'PulsarReady', 'status': 'True', 'type': 'PulsarReady'}, {'lastTransitionTime': '2022-05-07T07:54:48Z', 'message': 'All Milvus components are healthy', 'reason': 'MilvusClusterHealthy', 'status': 'True', 'type': 'MilvusReady'}], 'endpoint': '10.98.0.13:19530', 'ingress': {'loadBalancer': {}}, 'replicas': {'dataCoord': 1, 'dataNode': 2, 'indexCoord': 1, 'indexNode': 2, 'proxy': 1, 'queryCoord': 1, 'queryNode': 1, 'rootCoord': 1}, 'status': 'Healthy'}} (cus_resource_opts.py:59)
[2022-05-07 15:55:15,135 - INFO - ci_test]: start to check healthy: 1651910115.135755 (milvus_operator.py:125)
[2022-05-07 15:55:25,252 - INFO - ci_test]: milvus healthy in 10.117064952850342 seconds (milvus_operator.py:132)
[2022-05-07 15:55:25,444 - INFO - ci_test]: datacoord status is True (util_k8s.py:43)
[2022-05-07 15:55:25,444 - INFO - ci_test]: datanode status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: datanode status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: indexcoord status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: indexnode status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: proxy status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: querycoord status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: querynode status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: rootcoord status is True (util_k8s.py:43)
[2022-05-07 15:55:25,445 - INFO - ci_test]: all pods in namespace chaos-testing with label app.kubernetes.io/instance=shrink-index are ready (util_k8s.py:52)
[2022-05-07 15:55:25,447 - DEBUG - ci_test]: (api_request)  : [Collection.create_index] args: ['float_vector', {'index_type': 'IVF_SQ8', 'metric_type': 'L2', 'params': {'nlist': 128}}], kwargs: {'timeout': 40} (api_request.py:55)
[2022-05-07 15:55:26,080 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 1030, in create_index
    return conn.create_index(self._name, field_name, index_params,
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 625, in create_index
    index_success, fail_reason = self.wait_for_creating_index(collection_name=collection_name,
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 679, in wait_for_creating_index
    state, fail_reason = self.get_index_state(collection_name, field_name, timeout, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 671, in get_index_state
    raise MilvusException(status.error_code, status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=DescribeSegments failed: cannot find index, id = 433038323890257921)>
 (api_request.py:35)
[2022-05-07 15:55:26,081 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=DescribeSegments failed: cannot find index, id = 433038323890257921)> (api_request.py:36)
```
```


### Anything else?

_No response_

**Comment by yanliang567** [comment]
@bigsheeper could you please help on this issue?

/assign @bigsheeper 
/unassign

**Comment by xige-16** [comment]
I'm looking into this issue /assign
/unassign @bigsheeper

**Comment by xige-16** [comment]
same with #16699

**Comment by xige-16** [comment]
The reason is that the old index of some segments was not deleted when the index was dropped. When the index was created for the second time, the old index id was used to query the progress of segment build index, so the error "index id not found" was reported.

**Comment by xige-16** [comment]
The bug should has been fixed , please check it. @ThreadDao

**Comment by ThreadDao** [comment]
Fixed `master-20220509-76bfd419`
